### PR TITLE
Add upgrade steps to get the changes done in the ngeo application

### LINKED ISSUE #2276: Have a way to track the changes to do from the ngeo templates

### PR DESCRIPTION
**Pull request by sbrunner**
and optionally all other changes done in the create template.

Faster upgrade if the changelog is empty

fix #2276

### CODE REVIEW

File: c2cgeoportal/scripts/c2ctool.py [hunk 1]
```diff
@@ -220,6 +220,10 @@ def upgrade(self):
             self.step3()
         elif self.options.step == 4:
             self.step4()
+        elif self.options.step == 5:
+            self.step6()
```
Comment (pvalsecc) [inline c2cgeoportal/scripts/c2ctool.py]: I'm sure it was to test I'm doing proper reviews ;-)
`self.step5()`

File: c2cgeoportal/scripts/c2ctool.py [hunk 2]
```diff
@@ -344,32 +346,86 @@ def step1(self):
...
+                "Apply the manual migration steps based on what is in the CONST_CHANGELOG.txt file"
+                " (listed in the `changelog.diff` file)."
+            )
+            self.print_step(2)
 
     def step2(self):
+        if os.path.isfile("changelog.diff"):
+            os.unlink("changelog.diff")
+
+        diff_file = open("ngeo.diff", "w")
```
Comment (pvalsecc) [inline c2cgeoportal/scripts/c2ctool.py]: Just a comment, you don't have to act on it...

I prefere that construct (safer and more compact):
```python
with open("ngeo.diff", "w") as diff_file:
    ...
```

File: c2cgeoportal/scripts/c2ctool.py [hunk 3]
```diff
@@ -391,9 +447,9 @@ def step2(self):
             print("You are running on Windows, please restart your Apache server,")
             print("because we can not do that automatically.")
 
-        self.print_step(3)
+        self.print_step(5)
```
Comment (pvalsecc) [inline c2cgeoportal/scripts/c2ctool.py]: Why not doing the step5 automatically if not windows?

File: c2cgeoportal/scripts/c2ctool.py [hunk 4]
```diff
@@ -391,9 +447,9 @@ def step2(self):
...
 
-    def step3(self):
+    def step5(self):
         if not self.test_checkers():
             print("")
             self.print_step(3, intro="Correct the checker, the step 3 again:")
```
Comment (pvalsecc) [inline c2cgeoportal/scripts/c2ctool.py]: s/the step 3 again/then run the step 3 again/

File: c2cgeoportal/scripts/c2ctool.py (hunk 1 above)
Comment (sbrunner) [inline c2cgeoportal/scripts/c2ctool.py]: thanks :-)

File: c2cgeoportal/scripts/c2ctool.py (hunk 2 above)
Comment (sbrunner) [inline c2cgeoportal/scripts/c2ctool.py]: +1 :-)

File: c2cgeoportal/scripts/c2ctool.py (hunk 4 above)
Comment (sbrunner) [inline c2cgeoportal/scripts/c2ctool.py]: Thanks :-)

File: c2cgeoportal/scripts/c2ctool.py (hunk 3 above)
Comment (sbrunner) [inline c2cgeoportal/scripts/c2ctool.py]: To purpose the user to test the application manually :-)

**Comment by sbrunner** [comment]
Thanks @pvalsecc updated :-)

**Comment by pvalsecc** [comment]
You can merge. Thanks.

**Pull request merged**